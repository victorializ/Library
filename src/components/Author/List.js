import React from 'react';

import { List } from 'semantic-ui-react';
import { Author } from './Author';

const AuthorsList = (
  {
    data,
    onAuthorSelect
  }) => 
  <List inverted divided>
    { data.map(
      ({
        authorId, 
        firstName, 
        lastName
      }) => 
      <Author 
          key={authorId}
          authorId={authorId}
          firstName={firstName}
          lastName={lastName}
          onAuthorSelect={onAuthorSelect}
          className='authors__list-item'
      />
    )}
  </List>

export { AuthorsList };