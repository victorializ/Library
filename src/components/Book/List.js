import React  from 'react';

import { Book } from './Book';
import { ItemsList } from '../ItemsList';

const List = ({data}) => 
    <ItemsList 
        data={data} 
        Component={Book}
    />

export { List as BooksList };    
