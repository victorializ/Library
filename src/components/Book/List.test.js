import React from 'react';
import { shallow } from 'enzyme';

import { BooksList } from './List';

describe('BooksList component', () => {
    it('renders books list component correctly', () => {
        expect(shallow(<BooksList />)).toMatchSnapshot();
    });
});