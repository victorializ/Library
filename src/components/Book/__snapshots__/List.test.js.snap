// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksList component renders books list component correctly 1`] = `
<ItemsList
  Component={
    Object {
      "$$typeof": Symbol(react.memo),
      "WrappedComponent": [Function],
      "compare": null,
      "displayName": "Connect(Book)",
      "type": [Function],
    }
  }
/>
`;
